Docs: Interactivity API: Add wp_interactivity_state() clarification (#64356)
WordPress/gutenberg: docs/reference-guides/interactivity-api/api-reference.md

## Server functions

> The Interactivity API comes with handy functions that allow you to initialize and reference configuration options on the server. This is a great way to leverage many of WordPress's APIs, like nonces, AJAX, and translations.

Reviewer: ```suggestion
The Interactivity API comes with handy functions that allow you to initialize and reference configuration options on the server. This is necessary to feed the initial data that the Server Directive Processing will use to modify the HTML markup before it's send to the browser. It is also a great way to leverage many of WordPress's APIs, like nonces, AJAX, and translations.
```

const { showLikeButton } = getConfig();
```

### wp_interactivity_state

`wp_interactivity_state` allows the initialization of state on the server, which will be merged with with any stores defined in `view.js`.

Initializing state on the server allows you to use many critical WordPress APIs, including [AJAX](https://developer.wordpress.org/plugins/javascript/ajax/), or [nonces](https://developer.wordpress.org/plugins/javascript/enqueuing/#nonce).

The `wp_interactivity_state` function receives two arguments, a string with the namespace that will be used as a reference and an associative array containing the values.

Here is an example of passing the WP Admin AJAX endpoint with a nonce.

```php
// render.php

wp_interactivity_state(
	'myPlugin',
	array(
		'ajaxUrl' => admin_url( 'admin-ajax.php' ),
		'nonce'   => wp_create_nonce( 'myPlugin_nonce' ),
	),
);
```

```js
const { state } = store( 'myPlugin', {
	console.log( state.ajaxUrl );
	// Expected output: https://yoursite.com/wp-admin/admin-ajax.php
}

> ```

Reviewer: This code snippet is not correct because you're executing a console.log inside a JavaScript object.

Perhaps it would be interesting to put here a complete example (although simple) of how to use AJAX with the nonce?

```js
const { state } = store('myPlugin', {
  actions: {
    *doSomething() {
      try {
        const formData = new FormData();
        formData.append('action', 'do_something');
        formData.append('_ajax_nonce', state.nonce);

        const data = yield fetch(state.ajaxUrl, {
          method: 'POST',
          body: formData,
        }).then((response) => response.json());

        console.log('Server data!', data);
      } catch (e) {
        // Something went wrong!
      }
    },
  },
});
```

Author: Yes, totally appreciate the clarity here. I pushed an update to address. However, I prefer the expanded syntax (extra white-space), but I do not want to introduce a wide philsophical debate on ESLint and formatting. So, if this is not ideal formatting then let me know and I'm happy to push a different less-whitespace format. Thanks!

Reviewer: Yes, in the final file, it should adhere to the WordPress coding standard. If you have your IDE set up with a Prettier extension, it should format it automatically when you format the Markdown file.

_It is not formatted with spaces here because I am using the normal Prettier when writing my GH comments, not the WordPress fork._

Reviewer: Oh, I just realized that file wasn't formatted. Well, let's just merge this pull request and I'll create another one later to format it.

const { showLikeButton } = getConfig();
```

### wp_interactivity_state

> `wp_interactivity_state` allows the initialization of state on the server, which will be merged with with any stores defined in `view.js`.

Reviewer: ```suggestion
`wp_interactivity_state` allows the initialization of the global state on the server, which will be used to process the directives on the server and then will be merged with any global state defined in the client.
```

> Initializing state on the server allows you to use many critical WordPress APIs, including [AJAX](https://developer.wordpress.org/plugins/javascript/ajax/), or [nonces](https://developer.wordpress.org/plugins/javascript/enqueuing/#nonce).

Reviewer: ```suggestion
Initializing the global state on the server also allows you to use many critical WordPress APIs, including [AJAX](https://developer.wordpress.org/plugins/javascript/ajax/), or [nonces](https://developer.wordpress.org/plugins/javascript/enqueuing/#nonce).
```
